lib/src/util/dart.dart

```diff
@@ -78,3 +86,84 @@ SourceSpan contextualizeSpan(
 
   return file.span(start, contextRunes.offset);
 }
+
+class AnalysisSessionManager {
+  final ResourceProvider resourceProvider;
+  final List<AnalysisContext> _contexts = [];
+
+  AnalysisSessionManager({ResourceProvider resourceProvider})
+      : resourceProvider =
+            resourceProvider ?? PhysicalResourceProvider.INSTANCE;
+
+  /// Parse the file with the given [path] into AST.
+  CompilationUnit parse(String path) {
+    var parseResult = _getSession(path).getParsedAstSync(path);
+    if (parseResult.errors.isNotEmpty) {
+      throw new AnalyzerErrorGroup(parseResult.errors);
+    }
+    return parseResult.unit;
+  }
+
+  /// Return import and export directives in the file with the given [path].
+  List<UriBasedDirective> parseImportsAndExports(String path) {
+    var unit = parse(path);
+    var uriDirectives = <UriBasedDirective>[];
+    for (var directive in unit.directives) {
+      if (directive is UriBasedDirective) {
```

Comment:
This will include part directives. Is that intentional?

Reply:
No, actually we don't need it at all.
Removing.